File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        // This is 1ms solution below uses hashset which is 3ms solution look both
        int m=matrix.length; int n=matrix[0].length;
        boolean[] rows =new boolean[m];
        boolean[] cols =new boolean[n];
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j]==0){
                    rows[i]=true;
                    cols[j]=true;
                }
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(rows[i] || cols[j]){
                    matrix[i][j]=0;
                }
            }
        }



        // HashSet<Integer>rows=new HashSet<>();
        // HashSet<Integer>cols=new HashSet<>();
        // int m=matrix.length; int n=matrix[0].length;
        // for(int i=0;i<m;i++){
        //     for(int j=0;j<n;j++){
        //         if(matrix[i][j]==0){
        //             rows.add(i);
        //             cols.add(j);
        //         }
        //     }
        // }
        // for(int row:rows){
        //     for(int j=0;j<n;j++){
        //         matrix[row][j]=0;
        //     }
        // }
        // for(int col:cols){
        //     for(int j=0;j<m;j++){
        //         matrix[j][col]=0;
        //     }
        // }
  }
    
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # m,n=len(matrix) , len(matrix[0])
        # rows=set()
        # cols=set()

        # for i in range(m):
        #     for j in range(n):
        #         if matrix[i][j]==0:
        #             rows.add(i)
        #             cols.add(j)
        # for row in rows:
        #     for j in range(n):
        #         matrix[row][j]=0
        # for col in cols:
        #     for j in range(m):
        #         matrix[j][col]=0

        m = len(matrix) 
        n = len(matrix[0])
        rows = [False] * m 
        cols = [False] * n  
        for i in range(m): 
            for j in range(n): 
                if matrix[i][j] == 0: 
                    rows[i] = True
                    cols[j] = True 
        # Second loop: set the required rows and columns to zero 
        for i in range(m): 
            for j in range(n): 
                if rows[i] or cols[j]: 
                    matrix[i][j] = 0
        